src/parser/string.rs

```diff
@@ -6,8 +6,10 @@ use std::io::Read;
 use xml::reader::Events;
 use xml::reader::XmlEvent;
 
+use GpxVersion;
+
 /// consume consumes a single string as tag content.
-pub fn consume<R: Read>(reader: &mut Peekable<Events<R>>) -> Result<String> {
+pub fn consume<R: Read>(reader: &mut Peekable<Events<R>>, _version: GpxVersion) -> Result<String> {
```

Comment:
I feel as if this version passing becomes an antipattern, this file being a good example. It comes with lots of boiler plate code that, while consistent, isn't clean. What do you think of some sort of larger struct with version data as an alternative?

```rust
pub fn consume<R: Read>(context: &mut GpxContext<Peekable<Events<R>>>) -> Result<String> {
  for event in context.reader() {
    // ...
      if (context.version() == GpxVersion::11) { /* ... */ }
    // ...
  }
}
```

While the signature is ugly (`GpxContext<Peekable<Events<Read>>>`), I'd genuinely like to hear your opinion.

Reply:
How about `context: &mut Context<R: Read>` ? I have a patch doing that.